1-Draft/RFCNNNN-Native-Markdown-Rendering.md

```diff
@@ -0,0 +1,141 @@
+---
+RFC: 
+Author: Aditya Patwardhan
+Status: Draft
+SupercededBy: 
+Version: 0.1
+Area: Formatting and Output
+Comments Due: 03/26/2018
+Plan to implement: Yes
+---
+
+# Native Markdown Rendering
+
+Render markdown content on console to improve readability. 
+
+## Motivation
+
+Markdown is a common authoring format used by the community.
+There is no easy way in PowerShell to visualize a markdown document on console.
+
+## Specification
+
+This RFC proposes to use VT100 escape sequences to render markdown content.
+`ConvertFrom-Markdown` cmdlet as part of `Microsoft.PowerShell.MarkdownRender` PowerShell module would consume a string or a file path and output a VT100 encoded string.
+Optionally, the output can be formatted as HTML.
+For converting strings to VT100 coded strings, we will be writing an extension to [markdig](https://github.com/lunet-io/markdig).
+The extension will insert VT100 escape sequences as appropriate.
+
+```PowerShell
+ConvertFrom-Markdown [-Path] <string[]> -AsHTML -AsPlainText  [<CommonParameters>]
+
+ConvertFrom-Markdown -LiteralPath <string[]> -AsHTML -AsPlainText  [<CommonParameters>]
+
+ConvertFrom-Markdown -InputObject <psobject> -AsHTML -AsPlainText  [<CommonParameters>]
+```
+
+### Parameters
+
+- **Path** : Accepts an array of file paths with markdown content.
+- **LiteralPath** : Accepts an array of literal paths with markdown content.
+- **InputObject** : Accepts an InputObject of `System.IO.FileSystemInfo`, `string` type.
+- **AsHTML** : When selected, the output will be HTML.
+- **AsPlainText** : When selected, the output will be plain text.
+
+The default output will be a string encoded with VT100 escape sequences.
+
+## Supported Markdown Elements
+
+We will be supporting a limited set of markdown elements in the initial version.
+
+### Block Elements
+
+The block elements include paragraphs and line breaks, headers, block quotes, code blocks and horizontal rules.
+
+#### Paragraphs and line breaks
+
+These will be rendered as plain text.
+
+#### Headers
+
+VT100 escape sequences for headers are as follows:
```

Comment:
I think the escape sequences for different headers should be configurable, because users may have different background colors for their hosts, and some pre-defined color won't work well with that background.

Reply:
I believe we don't need to give the granularity that every markdown header color can be configured. Instead, we can provide 2 or 3 themes that would work for dark, light backgrounds respectively.